File: core/src/main/java/uk/lobsterdoodle/namepicker/events/EventBus.java
package uk.lobsterdoodle.namepicker.events;

public interface EventBus {

    void post(Object event);

    void postSticky(Object event);

    <T> T getStickyEvent(Class<T> c);

    void register(Object subscriber);

    void unregister(Object subscriber);

    void removeStickyEvent(Class clazz);

    boolean isRegistered(Object subscriber);
}

File: core/src/main/kotlin/uk/lobsterdoodle/namepicker/events/EventBus.kt
package uk.lobsterdoodle.namepicker.events

interface EventBus {

    fun post(event: Any)

    fun postSticky(event: Any)

    fun <T> getStickyEvent(clazz: Class<T>): T

    fun register(subscriber: Any)

    fun unregister(subscriber: Any)

    fun <T> removeStickyEvent(clazz: Class<T>)

    fun isRegistered(subscriber: Any): Boolean
}
